test/org/sagebionetworks/bridge/play/controllers/BaseControllerTest.java

```diff
@@ -350,7 +396,74 @@ public void canGetLanguagesWhenNotInSessionOrHeader() throws Exception {
+        // testing this because the rest of these tests will use ImmutableSet.of()
+        assertTrue(langs instanceof LinkedHashSet);
+        assertEquals(ImmutableSet.of(), langs);
+
+        // verify if it set warning header
+        Http.Response mockResponse = BaseController.response();
+        verify(mockResponse).setHeader(BRIDGE_API_STATUS_HEADER, WARN_NO_ACCEPT_LANGUAGE);
+    }
+
+    @Test
+    public void setWarnHeaderWhenEmptyAcceptLanguage() throws Exception {
+        BaseController controller = new SchedulePlanController();
+        mockPlayContext();
+        mockHeader(ACCEPT_LANGUAGE, "");
+
+        // with no accept language header at all, things don't break;
+        LinkedHashSet<String> langs = controller.getLanguagesFromAcceptLanguageHeader();
+        // testing this because the rest of these tests will use ImmutableSet.of()
+        assertTrue(langs instanceof LinkedHashSet);
+        assertEquals(ImmutableSet.of(), langs);
+
+        // verify if it set warning header
+        Http.Response mockResponse = BaseController.response();
+        verify(mockResponse).setHeader(BRIDGE_API_STATUS_HEADER, WARN_NO_ACCEPT_LANGUAGE);
+    }
+
+    @Test
+    public void setWarnHeaderWhenNullAcceptLanguage() throws Exception {
+        BaseController controller = new SchedulePlanController();
+        mockPlayContext();
+        mockHeader(ACCEPT_LANGUAGE, null);
+
+        // with no accept language header at all, things don't break;
+        LinkedHashSet<String> langs = controller.getLanguagesFromAcceptLanguageHeader();
+        // testing this because the rest of these tests will use ImmutableSet.of()
+        assertTrue(langs instanceof LinkedHashSet);
+        assertEquals(ImmutableSet.of(), langs);
+
+        // verify if it set warning header
+        Http.Response mockResponse = BaseController.response();
+        verify(mockResponse).setHeader(BRIDGE_API_STATUS_HEADER, WARN_NO_ACCEPT_LANGUAGE);
+    }
+
+    @Test
+    public void setWarnHeaderWhenInvalidAcceptLanguage() throws Exception {
+        BaseController controller = new SchedulePlanController();
+        mockPlayContext();
+        mockHeader(ACCEPT_LANGUAGE, "ThisIsAnVvalidAcceptLanguage");
+
+        // with no accept language header at all, things don't break;
+        LinkedHashSet<String> langs = controller.getLanguagesFromAcceptLanguageHeader();
+        // testing this because the rest of these tests will use ImmutableSet.of()
+        assertTrue(langs instanceof LinkedHashSet);
+        assertEquals(ImmutableSet.of(), langs);
+
+        // verify if it set warning header
+        Http.Response mockResponse = BaseController.response();
+        verify(mockResponse).setHeader(BRIDGE_API_STATUS_HEADER, WARN_NO_ACCEPT_LANGUAGE);
+    }
+
     @Test(expected = NotAuthenticatedException.class)
     public void getSessionAuthenticatedFail() {
         UserSession session = new UserSession(new StudyParticipant.Builder().build());
```

Comment:
Does the deprecated header test have to be changed? I don't think so if 1) you consolidate the code adding the header and 2) verify that messages are concatenated correctly.